File: src/main/java/com/samtang/Hello.java
package com.samtang;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello World");
//        new Person().Hello();

        String s = "abced";
        //s = null; //如果為 null, 下方取得的方法會出現編譯錯誤 Exception

        System.out.println(s.charAt(3));
        System.out.println(s.substring(3));
        System.out.println(s.substring(1,4));
        System.out.println(s.length());

        Person p = new Person("Tom",66.5f, 1.7f);
        p.Hello();

        System.out.println (p.bmi());

        int score = 88;
        System.out.println (score > 60);
        System.out.println (score > 60 && score < 90);
        System.out.println (score <60 || score >90);

        // char byte 變數因為都是 ASC整數碼, 所以可以做比較
        char c = 'A';
        System.out.println (c > 60);
        System.out.println (c > 'a');

        /*int age = 19;
        Integer age2 = 20;
        char c = 'I';
        byte b = 120;
        float weight = 66.5f;
        float height = 1.7f;
        boolean aduit = true;
        boolean enroll = false;
        String name = "Samtang";*/
    }
}

File: src/main/kotlin/com/samtang/Hello.kt
package com.samtang

fun main() {
//    println("Hello Kotlin")
//    Human().Hello()
    val h = Human(weight = 66.5f, height = 1.7f)
    h.Hello()
    println(h.bmi())

    val score = 88
    println(score > 80)

    val c : Char = 'A'
    println(c.toInt() > 60) //char 型態不能直接比較, 需要轉換成數值

   /* val age = 19 //val 宣告變數後無法更換變數值
    //age = 20  //指定為 val 重新塞值會錯誤

    var age2 : Int = 19 //val 宣告變數 : 後方可以指定型態
    age2 = 20   //可以重新塞值

    var weight = 22.6   //無需宣告型態 預設會 double 型態
    var height = 2.4f   //指定變數型能為 float

    var name : String   //若非一開始塞值給變數, 則必須需告變數型態
    name = "Samtang"*/

}

//建構子參數 name 給固定初始值, 就可以選擇性的代入 name 變數
class Human( var name : String = "", var weight : Float, var height : Float) {

    init {
        println("test $weight"); //等同於 println("test "+ weight)
    }

    fun bmi(): Float {
        val bmi = weight / (height * height)
        return bmi
    }

    fun Hello() {
        println("Hello Kotlin use Human")
    }
}